src/server/local-auth-helpers.js

```diff
@@ -0,0 +1,187 @@
+import AuthHasher from 'passport-local-authenticate'
+import { User, Invite, Organization } from './models'
+import { capitalizeWord } from './api/lib/utils'
+
+const errorMessages = {
+  invalidInvite: 'Invalid invite code. Contact your administrator.',
+  invalidCredentials: 'Invalid username or password',
+  emailTaken: 'That email is already taken.',
+  passwordsDontMatch: 'Passwords don\'t match.',
+  invalidResetHash: 'Invalid username or password reset link. Contact your administrator.',
+  noSamePassword: 'Old and new password can\'t be the same'
+}
+
+const validUuid = async (nextUrl, uuidMatch) => {
+  let foundUUID
+  if (nextUrl.includes('join')) {
```

Comment:
* this needs to fail fast if `!uuidMatch`  -- if someone submits a request without a validUuid at all, we don't even need to do a search.  
* Also let's return something in the non-error case so we can test affirmatively for that below.

Reply:
Instead of returning an array that can be used by the `done` callback, I am now throwing an error that is handled in the `LocalStrategy` constructor. I explain why in the commit body of my last commit. Curious to hear your opinion, and happy to change it back.